src/main/java/bio/terra/service/filedata/FileMetadataUtils.java

```diff
@@ -77,10 +80,33 @@ public FireStoreDirectoryEntry makeDirectoryEntry(String lookupDirPath) {
   public String makeLookupPath(String fullPath) {
     String temp = StringUtils.prependIfMissing(fullPath, "/");
     temp = StringUtils.removeEnd(temp, "/");
-    return ROOT_DIR_NAME + temp;
+    temp = StringUtils.prependIfMissing(temp, ROOT_DIR_NAME);
+    return temp;
   }
 
   public String makePathFromLookupPath(String lookupPath) {
     return StringUtils.removeStart(lookupPath, ROOT_DIR_NAME);
   }
+
+  public List<String> findNewDirectoryPaths(
+      List<FireStoreDirectoryEntry> datasetEntries, LRUMap<String, Boolean> pathMap) {
+
+    List<String> pathsToCheck = new ArrayList<>();
```

Comment:
Not sure why this code uses `List` instead of `Set`, as it looks like there's no benefit in checking the same path more than once. This would only be used if the number of paths exceeds the LRU cache size, which may be unlikely, but it seems like it would better express the intent of the code.

Reply:
As I mentioned in my comment, this is not new code, it's just moved to be shared between gcp and azure. I'll create a new ticket to re-factor and add tests for this method because I want to make sure I do a non-rushed job since this will impact production code (i.e. impacts both gcp, not just azure).